### DISCUSSION TITLE
Simulate Unavailability

### DISCUSSION THREAD

**Discussion by rahultaneja1534** [orlangure/gnomock#670] [Q&A]
Does `gnmock` provide capability to simulate temporary unavailability of the used component? I am currently using `RabbitMQ` and `MySQL` and would like to test how the my application behaves when one of these components aren't available for some time.

**Comment by orlangure** [comment]
Hi @rahultaneja1534,
There is currently no such built-in capability, but there are a few ways that I can think of to trigger failures:
- in mysql, a query can be executed from the test that changes the password of the user used in tests, and then kills their connection. Later, the password can be restored.
- the entire container can be killed mid-test if the state is not important
- in containers that have `exec`-able binaries inside, one could think about a command to execute to make the service temporary unavailable

I think that if you have any particular failure in mind, you could share your expectations, and we can think about a way to make it happen. In general I support the idea to extend the built-in features with failure simulation mechanisms wherever possible.
